### PR TITLE
Rename project to camel-case

### PR DESCRIPTION
**Pull request by dennisklein**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -16,19 +16,14 @@
 cmake_minimum_required(VERSION 3.15 FATAL_ERROR)
 cmake_policy(VERSION 3.15...3.20)
 
-# Set project version
-SET(FAIRROOT_MAJOR_VERSION 19)
-SET(FAIRROOT_MINOR_VERSION 0)
-SET(FAIRROOT_PATCH_VERSION 0)
+set(FairRoot_VERSION_MAJOR 19)
+set(FairRoot_VERSION_MINOR 0)
+set(FairRoot_VERSION_PATCH 0)
```
Comment (ChristianTackeGSI) [inline CMakeLists.txt:13]: Drop this completely.

**Review by ChristianTackeGSI — COMMENTED** [review]
1. I wanted to ask about a ChangeLog entry (about `FAIRROOT_VERSSION` not being exported by some cmake stuff anymore), but I think, the public API does not change anyway. So no request for a ChangeLog entry

2. Could you move the `templates/` changes into a dedicated commit?
   * The use of `FAIRROOT_LIBRARY_PROPERTIES` vs. `PROJECT_LIBRARY_PROPERTIES` is a public API of our `ROOTMacros.cmake`. So that's a different thing from doing the same on our "internal" CMakeLists.txt
   * Cleaning up whitespace feels very unrelated anyhow.

File: CMakeLists.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (dennisklein) [reply CMakeLists.txt]: done

File: CMakeLists.txt [hunk 2]
```diff
@@ -16,19 +16,14 @@
...
 set(PROJECT_MIN_CXX_STANDARD 17)
 
-# Set name of our project to "FAIRROOT".
-# Has to be done after check of cmake version
-# This also sets ${FAIRROOT_VERSION} to the provided VERSION value, which would be empty if not set here explicitly
-project(FAIRROOT VERSION ${FAIRROOT_MAJOR_VERSION}.${FAIRROOT_MINOR_VERSION}.${FAIRROOT_PATCH_VERSION})
-# where to look first for cmake modules, before ${CMAKE_ROOT}/Modules/
-# is checked
+project(FairRoot VERSION ${FairRoot_VERSION_MAJOR}.${FairRoot_VERSION_MINOR}.${FairRoot_VERSION_PATCH})
+
```
[the remark this answers is not in the document]
Comment (dennisklein) [reply CMakeLists.txt]: done

File: cmake/scripts/fairroot-config.in [hunk 3]
```diff
@@ -1,10 +1,10 @@
...
 
-_major_version=@FAIRROOT_MAJOR_VERSION@
-_minor_version=@FAIRROOT_MINOR_VERSION@
-_patch_version=@FAIRROOT_PATCH_VERSION@
+_major_version=@PROJECT_VERSION_MAJOR@
+_minor_version=@PROJECT_VERSION_MINOR@
+_patch_version=@PROJECT_VERSION_PATCH@
 
-_version=v@FAIRROOT_MAJOR_VERSION@.@FAIRROOT_MINOR_VERSION@.@FAIRROOT_PATCH_VERSION@
+_version=v@PROJECT_VERSION_MAJOR@.@PROJECT_VERSION_MINOR@.@PROJECT_VERSION_PATCH@
```
[the remark this answers is not in the document]
Comment (dennisklein) [reply cmake/scripts/fairroot-config.in]: done

File: CMakeLists.txt [hunk 4]
```diff
@@ -349,12 +331,12 @@ set (Git_VERSION ${GIT_VERSION_STRING})
...
 If(GIT_FOUND AND EXISTS "${CMAKE_SOURCE_DIR}/.git")
   Execute_Process(COMMAND ${GIT_EXECUTABLE} describe --tags
-                  OUTPUT_VARIABLE FAIRROOT_GIT_VERSION
+                  OUTPUT_VARIABLE PROJECT_GIT_VERSION
                   OUTPUT_STRIP_TRAILING_WHITESPACE
                   WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
                  )
 Else()
-  Set(FAIRROOT_GIT_VERSION v${FAIRROOT_MAJOR_VERSION}.${FAIRROOT_MINOR_VERSION}.${FAIRROOT_PATCH_VERSION})
+  set(PROJECT_GIT_VERSION v${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH})
```
Comment (ChristianTackeGSI) [inline CMakeLists.txt:339]:
```suggestion
  set(PROJECT_GIT_VERSION "v${PROJECT_VERSION}")
```

Maybe?

**Review by ChristianTackeGSI — APPROVED** [review]
Looks very good. I like a lot of this.

File: CMakeLists.txt (hunk 4 above)
Comment (dennisklein) [reply CMakeLists.txt]: :+1: done

Comment (dennisklein) [reply CMakeLists.txt]: also rebased to latest `dev` branch

**Review by ChristianTackeGSI — APPROVED** [review]

File: templates/project_root_containers/MyProjGenerators/rename.sh [hunk 5]
```diff
@@ -59,9 +59,8 @@ find . -name "*.h" -exec sed -e "s/FairStack/${Prefix}Stack/g" $sedstring "{}" "
 sed -e "s#tutorial/PndPythia#$RelativeDir#g" $sedstring CMakeLists.txt
 sed -e "s/PndPythia/$DetectorName/g" $sedstring CMakeLists.txt
 sed -e "s/PndPythia/$DetectorNameUpper/g" $sedstring CMakeLists.txt
-sed -e "s/FAIRROOT_SOURCE_DIR/$ProjectSourceDir/g" $sedstring CMakeLists.txt
 
-if [ -d .svn ]; then
```
Comment (fuhlig1) [inline templates/project_root_containers/MyProjGenerators/rename.sh:64]: Probably can be removed. This comes from a time when we were using Subversion

File: templates/project_stl_containers/MyProjGenerators/rename.sh [hunk 6]
```diff
@@ -59,9 +59,8 @@ find . -name "*.h" -exec sed -e "s/FairStack/${Prefix}Stack/g" $sedstring "{}" "
 sed -e "s#tutorial/PndPythia#$RelativeDir#g" $sedstring CMakeLists.txt
 sed -e "s/PndPythia/$DetectorName/g" $sedstring CMakeLists.txt
 sed -e "s/PndPythia/$DetectorNameUpper/g" $sedstring CMakeLists.txt
-sed -e "s/FAIRROOT_SOURCE_DIR/$ProjectSourceDir/g" $sedstring CMakeLists.txt
 
-if [ -d .svn ]; then  
+if [ -d .svn ]; then
   echo "Please remove the .svn directory."
   echo " This directory was also copied from templates."
   echo "##"
```
Comment (fuhlig1) [inline templates/project_stl_containers/MyProjGenerators/rename.sh:66]: Also here the block can be removed.

**Review by fuhlig1 — COMMENTED** [review]
You can also merge without removing the two blocks mentioned in my comments but since it is obsolete code I would propose to remove them.

File: templates/project_root_containers/MyProjGenerators/rename.sh (hunk 5 above)
Comment (dennisklein) [reply templates/project_root_containers/MyProjGenerators/rename.sh:64]: done

File: templates/project_stl_containers/MyProjGenerators/rename.sh (hunk 6 above)
Comment (dennisklein) [reply templates/project_stl_containers/MyProjGenerators/rename.sh]: done

**Review by fuhlig1 — APPROVED** [review]

**Review by ChristianTackeGSI — APPROVED** [review]

**Comment by fuhlig1** [comment]
Sorry I was too fast. In GitLab I am only able to merge if the complete pipeline succeeds.

**Comment by ChristianTackeGSI** [comment]
We could configure github so that the complete CI run needs to succeed. But as some checks are still not 100% stable, we haven't configured it that way and hope for people to actually look at the results.

**Comment by fuhlig1** [comment]
I will try to keep it in mind before doing the next merge.

**Pull request merged**